### PR TITLE
feat: add passkey loading state

### PR DESCRIPTION
**Pull request by moldy530**
# Pull Request Checklist

- [ ] Did you add new tests and confirm existing tests pass? (`yarn test`)
- [ ] Did you update relevant docs? (docs are found in the `site` folder, and guidelines for updating/adding docs can be found in the [contribution guide](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md))
- [ ] Do your commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] Does your PR title also follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] If you have a breaking change, is it [correctly reflected in your commit message](https://www.conventionalcommits.org/en/v1.0.0/#examples)? (e.g. `feat!: breaking change`)
- [ ] Did you run lint (`yarn lint:check`) and fix any issues? (`yarn lint:write`)
- [ ] Did you follow the [contribution guidelines](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md)?

---

## PR-Codex overview
This PR introduces improvements to the authentication flow in the Alchemy app, including passkey verification and error handling.

### Detailed summary
- Added passkey verification functionality
- Updated button types in `ButtonProps`
- Modified color variants in the theme
- Enhanced error display with `ErrorContainer`
- Refactored authentication context to `AuthStep`
- Implemented `usePasskeyVerify` hook
- Created `AddPasskey` component for adding passkeys

> The following files were skipped due to too many changes: `packages/alchemy/src/react/components/auth/card/loading/passkey.tsx`, `packages/alchemy/src/react/components/auth/sections/EmailAuth.tsx`, `packages/alchemy/src/react/components/auth/card/index.tsx`, `packages/alchemy/src/react/icons/passkey.tsx`, `packages/alchemy/src/react/components/auth/card/loading/email.tsx`, `packages/alchemy/src/react/context.tsx`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by moldy530** [comment]
image
image

File: packages/alchemy/src/react/components/auth/card/loading/email.tsx [hunk 1]
```diff
@@ -2,15 +2,21 @@ import { useAuthenticate } from "../../../../hooks/useAuthenticate.js";
...
+  context: Extract<AuthStep, { type: "email_verify" }>;
 }
 
 // eslint-disable-next-line jsdoc/require-jsdoc
 export const LoadingEmail = ({ context }: LoadingEmailProps) => {
-  const { authenticate } = useAuthenticate();
+  const { setAuthStep } = useAuthContext();
+  const { authenticate } = useAuthenticate({
+    onSuccess: () => {
+      // TODO: we will need to check if the dev has configured the app to allow for passkey as ssecondeary
```
Comment (dphilipson) [inline packages/alchemy/src/react/components/auth/card/loading/email.tsx:16]: Typo: "ssecondeary" lol

File: packages/alchemy/src/react/components/auth/context.ts [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
+  | { type: "email_verify"; email: string }
+  | { type: "passkey_verify"; error?: Error }
+  | { type: "initial" }
+  | { type: "complete" };
 
 type AuthContextType = {
-  authContext?: AuthContext;
-  setAuthContext: (context: AuthContext | undefined) => void;
+  authStep: AuthStep;
+  setAuthStep: (context: AuthStep) => void;
```
Comment (dphilipson) [inline packages/alchemy/src/react/components/auth/context.ts:11]: (Nit) Rename param `context` -> `step`.

File: packages/alchemy/src/react/components/auth/hooks/usePasskeyVerify.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+      setAuthStep({ type: "passkey_verify", error: err });
+    },
+    onSuccess: () => {
+      setAuthStep({ type: "complete" });
+    },
+  });
+
+  return {
+    isPending,
+    authenticate: () => authenticate({ type: "passkey", createNew: false }),
```
Comment (dphilipson) [inline packages/alchemy/src/react/components/auth/hooks/usePasskeyVerify.ts:27]: As an improvement, we might want to think about using `useCallback` functions we return to the SDK user. Some devs may have their apps set up to attempt to minimize re-renders using `memo` and returning a new function instance on each call to `usePasskeyVerify` could mess them up.

This comment assumes that `useAuthenticate` is returning the same `authenticate` instance every time. It's possible that implementing this suggestion codebase-wide would end up being fairly involved, and doesn't need to be done in this PR.

Comment (moldy530) [reply packages/alchemy/src/react/components/auth/hooks/usePasskeyVerify.ts:27]: ah yea `useAuthenticate` does return the same instance (as long as react-query does). 

I can update this authenticate method to be wrapped in `useCallback` based on that assumption

**Pull request merged**